src/mission_start.cpp

```diff
@@ -1627,3 +1627,92 @@ void mission_start::ranch_bartender_4(mission *miss)
 void mission_start::place_book( mission *)
 {
 }
+
+const tripoint reveal_destination ( const std::string& type, bool random )
+{
+    const tripoint your_pos = g->u.global_omt_location();
+    const tripoint center_pos = random ?
+    overmap_buffer.find_random( your_pos, type, rng( 40, 80 ), false ) :
+    overmap_buffer.find_closest( your_pos, type, rng ( 40, 80 ), false );
+
+    if( center_pos != overmap::invalid_tripoint ) {
+        overmap_buffer.reveal( center_pos, 2 );
+        return center_pos;
+    }
+
+    return overmap::invalid_tripoint;
+}
+
+void reveal_route( mission *miss, const tripoint destination )
+{
+    const tripoint source = g->u.global_omt_location();
+
+    const tripoint source_road = overmap_buffer.find_closest( source, "road", 3, false );
+    const tripoint dest_road = overmap_buffer.find_closest( destination, "road", 3, false );
+
+    if( overmap_buffer.reveal_route( source_road, dest_road ) )
```

Comment:
Missing/misplaced curly braces (here and below). See [CODE_STYLE.txt](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CODE_STYLE.txt) for details.